start_stop.lua

```diff
@@ -172,11 +121,13 @@ function courseplay:start(self)
 		self.cp.driver:delete()
 		self.cp.driver = UnloadableFieldworkAIDriver.create(self)
 	end
-	StartStopEvent:sendStartEvent(self)
+	---Make sure the clients and the server have the same farmId
+	local farmIndex = self.spec_enterable.controllerFarmId or self:getOwnerFarmId()
+	courseplay.onStartCpAIDriver(self,nil, false, farmIndex)
```

Comment:
What is unclear to me is how do we decided what to move into onStartCpAIDriver and what leave in :start()? Same for stop().

Reply:
I mainly created onStartCpAIDriver/onStopCpAIDriver to clear up the start_stop.lua and improve multiplayer code.
I sorted the code in courseplay.onStartCpAIDriver and courseplay.onStopAIDriver by giants code, which is also needed in the default Helper and cp code. Import is 'farmIndex' here and the 'helperIndex' in courseplay:onWriteStreamAIVehicle() which need to be the same on the server and client. Might be a god idea to override: 'AIVehicle:startAIVehicle()' with: `function courseplay:onStartCpAIDriver()
   if self:isCourseplayDriving() then 
      ...
  else 
   superFunc(self)
  end
` and the same for 'AIVehicle:stopAIVehicle()' but that has a few other limitations.